modules/dag-l1/src/main/scala/io/constellationnetwork/dag/l1/domain/address/storage/AddressStorage.scala

```diff
@@ -4,8 +4,10 @@ import io.constellationnetwork.schema.address.Address
 import io.constellationnetwork.schema.balance.Balance
 
 trait AddressStorage[F[_]] {
-  def getState: F[Map[Address, Balance]]
+  def getState: F[(Map[Address, Balance], Map[Address, Balance])]
   def getBalance(address: Address): F[Balance]
+  def getTokenLockBalance(address: Address): F[Balance]
```

Comment:
Why do you couple token locks with balances? 🤔 There should be a single balance only, and token locks should lock or unlock tokens, so the balance should be deducted or added

Reply:
We have 2 maps of balances now, the older balance and the new map with the tokenLockBalances. I'm not sure I'm understanding your question

Reply:
I think we've decided not to make the second balance map 🤔